src/librustc_typeck/check/compare_method.rs

```diff
@@ -374,7 +374,8 @@ fn compare_predicate_entailment<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
                 &infcx.parameter_environment.caller_bounds);
             infcx.resolve_regions_and_report_errors(&free_regions, impl_m_body_id);
         } else {
-            let fcx = FnCtxt::new(&inh, tcx.types.err, impl_m_body_id);
+            let mut fcx = FnCtxt::new(&inh, impl_m_body_id);
```

Comment:
Huh. It feels weird to me to not have the `ret_ty` get passed into the constructor. I see that we (one time) reassigned it later, but now it seems like we almost always do. I think I'd rather we continue to pass in a `Option<>` into `new` as an argument.